src/main/java/pl/edu/mimuw/logic/Formula.java

```diff
@@ -1,8 +1,11 @@
 package pl.edu.mimuw.logic;
 
+import java.util.List;
+
 public interface Formula {
 
   boolean eval(Valuation valuation);
 
-  // TODO: extend interface with needed methods
+  List<Variable> getAllVariables();
```

Comment:
ogólnie ok, polecam jednak używać kolekcji, które bedą mówiły same za siebie - jak użyjesz `Set<Variable>` to mówisz, że na pewno elemnty tam się nie powtórzą, a o coś takiego chodzi myślę